File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> h1= new HashMap<Integer, Integer>();
        int[] ans = new int[2];
        for(int i =0; i<nums.length; i++){
            int val = target -nums[i];
            if(!h1.containsKey(val))
                h1.put(nums[i], i);
            else{
                ans[0]=i;
                ans[1]= h1.get(val);
                return ans;
            }
        }
        return ans;
        
    }
}

File: 0001-two-sum/0001-two-sum.scala
import scala.collection.mutable.HashMap

object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    val h1 = HashMap[Int, Int]()
    val ans = new Array[Int](2)
    for (i <- 0 until nums.length) {
      val value = target - nums(i)
      if (!h1.contains(value)) {
        h1.put(nums(i), i)
      } else {
        ans(0) = i
        ans(1) = h1(value)
        return ans
      }
    }
    ans
  }
}
